tests/IResearch/ClusterCommMock.cpp

```diff
@@ -61,29 +61,37 @@ arangodb::OperationID ClusterCommMock::asyncRequest(
    bool singleRequest,
    arangodb::ClusterCommTimeout initTimeout
 ) {
-  auto entry = _requests.emplace(
-    std::piecewise_construct,
-    std::forward_as_tuple(++_operationId), // non-zero value
-    std::forward_as_tuple(coordTransactionID, destination, reqtype, path, body, headerFields, callback, singleRequest)
-  );
+  // execute before insertion to avoid consuming an '_operationId'
+  if (arangodb::rest::RequestType::PUT == reqtype
+      && std::string::npos != path.find("/_api/aql/shutdown/")) {
```

Comment:
Do this affect any of the AQL catch tests?

Reply:
it is required for the new CATCH tests that were added since the 'shutdown' requests ended up consuming all the test responses (that were meant to match subsequent requests) hence the mock code forcefully terminates such requests